homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -65,53 +76,97 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
+        """Fetch data from API endpoint."""
+        data = {}
+        async with async_timeout.timeout(30):
+            try:
+                await envoy_reader.getData()
+            except httpx.HTTPError as err:
+                raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+            for condition in monitored_conditions:
+                if condition != "inverters":
+                    data[condition] = await getattr(envoy_reader, condition)()
+                else:
+                    try:
+                        data["inverters_production"] = await getattr(
+                            envoy_reader, "inverters_production"
+                        )()
+                    except httpx.HTTPStatusError as err:
+                        _LOGGER.error("Authentication error: %s", err)
+                        data["inverters_production"] = None
+                        continue
+                    except httpx.HTTPError as err:
+                        _LOGGER.error("Protocol Error: %s", err)
+                        data["inverters_production"] = None
+                        continue
+
+            _LOGGER.debug("Retrieved data from API: %s", data)
+
+            return data
+
+    coordinator = DataUpdateCoordinator(
+        homeassistant,
+        _LOGGER,
+        name="sensor",
+        update_method=async_update_data,
+        update_interval=SCAN_INTERVAL,
+    )
+
+    await coordinator.async_refresh()
+
+    if coordinator.data is None:
+        raise PlatformNotReady
+
     entities = []
-    # Iterate through the list of sensors
     for condition in monitored_conditions:
-        if condition == "inverters":
-            try:
-                inverters = await envoy_reader.inverters_production()
-            except requests.exceptions.HTTPError:
-                _LOGGER.warning(
-                    "Authentication for Inverter data failed during setup: %s",
-                    ip_address,
-                )
-                continue
-
-            if isinstance(inverters, dict):
-                for inverter in inverters:
-                    entities.append(
-                        Envoy(
-                            envoy_reader,
-                            condition,
-                            f"{name}{SENSORS[condition][0]} {inverter}",
-                            SENSORS[condition][1],
-                        )
+        if (
+            condition == "inverters"
+            and coordinator.data.get("inverters_production") is not None
+        ):
+            for inverter in coordinator.data["inverters_production"]:
+                entities.append(
+                    Envoy(
+                        condition,
+                        f"{name}{SENSORS[condition][0]} {inverter}",
+                        SENSORS[condition][1],
+                        coordinator,
                     )
-
-        else:
+                )
+                _LOGGER.debug(
```

Comment:
Just make one logging call at the end and pass all entities to the log message. They have a repr magic method. Ie, don't log in the for loop.

Reply:
I removed this debug logging as it's not really needed.